plugins/database/src/db_plugin.cpp

```diff
@@ -5091,6 +5092,43 @@ irods::error db_reg_zone_op(
         return PASS( ret );
     }
 
+    // =-=-=-=-=-=-=-
+    // validate the connection string is well formed
+    const std::size_t addr_len = std::strlen(_zone_conn_info);
+    if (addr_len > 0) {
+        rodsHostAddr_t addr{};
+        // _zone_conn_info is const, so copy it
+        auto addr_buf = std::make_unique<char[]>(addr_len + 1);
+        std::strncpy(addr_buf.get(), _zone_conn_info, addr_len + 1);
```

Comment:
You can capture the `addr_len + 1` in a variable to guard against these lines becoming out of sync.

Alternatively, you can use a `std::vector` and avoid the size variable altogether.
```cpp
std::vector<char> addr_buf(addr_len + 1);
std::strncpy(addr_buf.data(), _zone_conn_info, addr_buf.size());
```